flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/AggregateITCase.scala

```diff
@@ -425,6 +426,80 @@ class AggregateITCase(
     assertEquals(expected, sink.getRetractResults)
   }
 
+  @Test
+  def testPrecisionForSumWithRetractAggregationOnDecimal(): Unit = {
+    val upsertSourceCurrencyData = List(
+      changelogRow("+I", 1.03520274.bigDecimal, 12345.035202748654.bigDecimal,
+        12.345678901234567.bigDecimal, "a"),
+      changelogRow("+I", 1.03520274.bigDecimal, 12345.035202748654.bigDecimal,
+        12.345678901234567.bigDecimal, "b"),
+      changelogRow("-D", 1.03520274.bigDecimal, 12345.035202748654.bigDecimal,
+        12.345678901234567.bigDecimal, "b"),
+      changelogRow("+I", 2.13520275.bigDecimal, 21245.542202748654.bigDecimal,
+        242.78594201234567.bigDecimal, "a"),
+      changelogRow("+I", 1.11111111.bigDecimal, 11111.111111111111.bigDecimal,
+        111.11111111111111.bigDecimal, "b"),
+      changelogRow("+I", 1.11111111.bigDecimal, 11111.111111111111.bigDecimal,
+        111.11111111111111.bigDecimal, "a"),
+      changelogRow("-D", 1.11111111.bigDecimal, 11111.111111111111.bigDecimal,
+        111.11111111111111.bigDecimal, "b"),
+      changelogRow("+I", 2.13520275.bigDecimal, 21245.542202748654.bigDecimal,
+        242.78594201234567.bigDecimal, "a"))
+
+    val upsertSourceDataId = registerData(upsertSourceCurrencyData);
+    tEnv.executeSql(
+      s"""
+         |CREATE TABLE T (
+         | `a` DECIMAL(32, 8),
+         | `b` DECIMAL(32, 20),
+         | `c` DECIMAL(32, 20),
+         | `d` STRING
+         |) WITH (
+         | 'connector' = 'values',
+         | 'data-id' = '${upsertSourceDataId}',
+         | 'changelog-mode' = 'I,D',
+         | 'failing-source' = 'true'
+         |)
+         |""".stripMargin)
+
+    val sql = "SELECT sum(a), sum(b), sum(c) FROM T GROUP BY d"
+
+    val sink = new TestingRetractSink
+    tEnv.sqlQuery(sql).toRetractStream[Row].addSink(sink)
+    env.execute()
+
+    val expected = List("6.41671935,65947.23071935707000000000,609.02867403703699700000")
```

Comment:
can you add a comment to explain the result? for me as a reviewer that is not super familiar with the topic anymore, it is difficult to see something in those numbers